### DISCUSSION TITLE
Actions intermittent errors

### DISCUSSION THREAD

**Discussion by rainyuanhx** [hasura/graphql-engine#10490] [General]
### Version Information

Server Version: v2.40.0
CLI Version (for CLI related issue):

### Environment
Hi My Friend

I encountered an error while using Hasura(v2.40.0) and I'm not sure where the problem is. Can you provide help?

The story goes like this.
I have a simple backend REST API interface that provides a GET method. To call this API program, I defined a new Action on HASURA with the following Query Type:
```
type Query {
  searchMonitor(businessId: String!): SearchMonitorResponse
}

type SearchMonitorResponse {
  test: String
  test2: String
  test3: String
}
```

Then, I called the GraphQL API on Postman
```
query MyQuery {
  searchMonitor(businessId: "abc") {
    test
  }
}
```

But sometimes I get the right results, 
![Catch887A(07-12-16-28-03)](https://github.com/user-attachments/assets/805ad613-a3d6-41a9-b92c-baf4b6ae9a10)

and sometimes I get error prompts
![Catch(07-12-16-28-03)](https://github.com/user-attachments/assets/5f1bef72-117e-4613-8e4c-348553b70558)

If you could provide an answer, I would greatly appreciate it.

**Comment by manasag** [comment]
This error is directly from the JSON library that is trying to decode the response body. Overall it indicates that the API response is not a valid json string. We haven't seen this issue reported elsewhere (and this code haven't been touched since some time), so I would recommend to debug the API endpoint for any issues.
Can you also check if any response transformation rules that have been setup for your action?
